ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,278 @@
+        ordered_product0.save_ordered_product()
+        quantity_before_change = OrderedProduct.objects.get(id=ordered_product0.id).quantity
+        response = client.get('/ordered_product/orders/'+str(ordered_product0.id), {'quantity': 5})
+        quantity_after_change = OrderedProduct.objects.get(id=ordered_product0.id).quantity
+        assert quantity_after_change == quantity_before_change+5
+        assert response.status_code == 302
+        assert response['Location'] == '/client/'
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_delete_order(self, client, saved_client0, ordered_product0, client0_login):
+        assert ordered_product0.user_name == saved_client0
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.filter(user_name=saved_client0)
+        response = client.get('/ordered_product/orders/'+str(ordered_product0.id)+',delete')
+        assert ordered_product0 not in OrderedProduct.objects.filter(user_name=saved_client0)
+        assert response.status_code == 302
+        assert response['Location'] == '/client/'
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_display_new_order_page(self, client, client0_login):
+        response = client.get('/ordered_product/new_order/')
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+
+    @pytest.mark.django_db()
+    def test_search_product_by_product_name(self, client, saved_supplier_product0, client0_login):
+        VALID_PRODUCT_NAME = saved_supplier_product0.qr_code.product_name
+        product = Product.objects.get(product_name=VALID_PRODUCT_NAME)
+        supplier_products_by_objects_filter = SupplierProduct.objects.filter(qr_code=product)
+        assert saved_supplier_product0 in supplier_products_by_objects_filter
+        response = client.get(
+            '/ordered_product/new_order/search_product_by_product_name',
+            {'product_name': VALID_PRODUCT_NAME}
+        )
+        assert response.status_code == 200
+        assert response.context['message'] == ""
+        supplier_products_by_the_web_view = response.context['supplier_products']
+        assert list(supplier_products_by_the_web_view) == list(supplier_products_by_objects_filter)
+
+    @pytest.mark.django_db()
+    def test_search_product_by_location(
+        self,
+        client,
+        saved_supplier_product0,
+        saved_client0,
+        delivery_location2,
+        client0_login
+    ):
+        delivery_location2.save()
+        assert delivery_location2.location == saved_client0.area
```

Comment:
I personally think those assertions are redundant, as the fixtures you create should be trusted, you hard-coded them. 
but if you feel more comfortable keeping them, I won't argue.

Reply:
I prefer to keep it beacause someone might edit the fixture and than this test will fail.
So it will be clear why the test failed.

Reply:
But that's ok, If someone decided to change a fixture, it is his responsability to notice what it changes and where, and in case there are real issues with his change, the test will fail anyway, alerting him he did something wrong.
Again, I don't mind keeping it, but I do want to make sure you understand the scope of tests and reaponsability you have once you need to edit tests or their data